compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -645,6 +645,10 @@ private boolean isAssignable(BType source, BType target, Set<TypePair> unresolve
         int sourceTag = source.tag;
         int targetTag = target.tag;
 
+        if (isNeverTypeOrStructureTypeWithARequiredNeverMember(source) && isAssignable(target, symTable.anyType)) {
```

Comment:
Do we even need to check the target type?

Will this conflict with your other PR btw?

Reply:
For any type `T`, the type `T|never` is the same as `T`. Therefore the target type can not be `error` type right? Therefore we do need to check the target type?

Nope. This check won't conflict with the other PR.

Reply:
We don't need the target type it seems.
Yeah. then this check will conflict with the other PR. will update the other to use the same